q2_assembly/spades/spades.py

```diff
@@ -114,11 +124,36 @@ def _assemble_spades(seqs, meta, common_args) -> ContigSequencesDirFmt:
     manifest = seqs.manifest.view(pd.DataFrame)
     result = ContigSequencesDirFmt()
 
-    for samp in list(manifest.index):
-        fwd = manifest.loc[samp, "forward"]
-        rev = manifest.loc[samp, "reverse"] if paired else None
+    if coassemble:
+        print(
+            "WARNING: The co-assembly process includes concatenating contents of all"
+            "read files. This process may take a while to finish!"
+        )
+        fwds = manifest["forward"]  # this will be  a list
+        revs = (
+            manifest["reverse"] if paired else None
+        )  # this will be a list if it exists
+
+        extension = get_file_extension(fwds[0])
+        fwd = f"all_samples_fwd{extension}"
+        rev = f"all_samples_rev{extension}" if paired else None
+
+        concatenate_files(fwds, fwd)
+        if paired:
+            concatenate_files(revs, rev)
 
-        _process_sample(samp, fwd, rev, common_args, result)
+        _process_sample("all_samples_spades", fwd, rev, common_args, result)
+
+        # delete the concatenated files created, after the assembly process is done
+        os.remove(fwd)
+        os.remove(rev)
```

Comment:
The issue with doing it this way is that the new, concatenated files will be generated in the "current" working directory, which will depend on how the user runs the action. To go around this, what we usually do is to create a temporary directory and place all the required files there. The added benefit of that is that you don't need to take care of cleaning up any files - when your process finishes they will be deleted automatically. You can see here for an example: https://github.com/bokulich-lab/q2-moshpit/blob/137eff9206f63aac2d79e45b50befd26918a5a6e/q2_moshpit/kaiju/classification.py#L232-L255